### PR TITLE
Add Model Class Diagram

### LINKED ISSUE #178: Update DG - Model Class Diagram

### PR DESCRIPTION
**Pull request by juliusgambe**
Screenshot 2023-10-26 at 3 51 04 PM
Screenshot 2023-10-26 at 3 51 13 PM

Screenshot 2023-10-26 at 3 51 18 PM

Screenshot 2023-10-26 at 3 51 23 PM

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -101,7 +101,8 @@ The `UI` component,
 
 ### Logic component
 
-**API** : [`Logic.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/logic/Logic.java)
+**
+API** : [`Logic.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/logic/Logic.java)
```
Comment (B-enguin) [inline docs/DeveloperGuide.md]: Should we change this to our own repo

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -142,35 +143,33 @@ How the parsing works:
 
 ### Model component
 
-**API** : [`Model.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/model/Model.java)
+**
+API** : [`Model.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/model/Model.java)
```
Comment (B-enguin) [inline docs/DeveloperGuide.md]: Same as above

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -142,35 +143,33 @@ How the parsing works:
...
-  is exposed to outsiders as an unmodifiable `ObservableList<Person>` that can be 'observed' e.g. the UI can be bound to
-  this list so that the UI automatically updates when the data in the list change.
+* stores the address book data i.e., all `Customer` objects (which are contained in a `UniqueCustomerList` object). The
+  address book is exposed to the outside as a `ReadOnlyBook` objects.
+* stores the delivery book data i.e., all `Delivery` objects (which are contained in a `UniqueDeliveryList` object). The
+  delivery book is exposed to the outside as a `ReadOnlyBook` objects.
+* stores the currently filtered `Customer` objects (e.g., results of a search query) as a separate _filteredCustomers_
+  list
+* stores the currently filtered `Delivery` objects (e.g., results of a status filter query) as a separate
+  _filteredDeliveries_ list
```
Comment (B-enguin) [inline docs/DeveloperGuide.md:159]: This would include find after #196

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -142,35 +143,33 @@ How the parsing works:
...
-
-<puml src="diagrams/BetterModelClassDiagram.puml" width="450" />
-
-</box>
 
 ### Storage component
 
-**API** : [`Storage.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/storage/Storage.java)
+**
+API** : [`Storage.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/storage/Storage.java)
```
Comment (B-enguin) [inline docs/DeveloperGuide.md]: Same as above, but already patched in #151

**Review by B-enguin — COMMENTED** [review]
Some general comments

**Review by jianyangg — APPROVED** [review]
LGTM!

**Pull request merged**